rich/_wrap.py

```diff
@@ -1,13 +1,16 @@
+from __future__ import annotations
+
 import re
-from typing import Iterable, List, Tuple
+from typing import Iterable
 
 from ._loop import loop_last
 from .cells import cell_len, chop_cells
 
 re_word = re.compile(r"\s*\S+\s*")
 
 
-def words(text: str) -> Iterable[Tuple[int, int, str]]:
+def words(text: str) -> Iterable[tuple[int, int, str]]:
```

Comment:
The first time I read the function name & docstring, I immediately expected the function to take a string like

```py
"Hey    how     are   you"
#0123456789012345678901234
```

And produce something like `(0, 2, "Hey"), (7, 9, "how"), ...` so I didn't even bother checking the regex pattern.
This, in turn, made it confusing to read `divide_line`.
So, all I'm saying is that maybe “Yields each word” is a bad start of the docstring because I don't know many people that would call the string `"Hey    "` a word.

Maybe mention that surrounding whitespace also gets consumed?
Or reword it to say that `words` breaks a string on non-whitespace characters or something like that?

Not sure if you get what I'm trying to say.